state-chain/chains/src/eth.rs

```diff
@@ -83,13 +80,43 @@ impl EthereumTrackedData {
 			.saturating_mul_int(self.base_fee)
 			.saturating_add(self.priority_fee)
 	}
+
+	pub fn calculate_ccm_gas_limit(
+		&self,
+		is_native_asset: bool,
+		gas_budget: GasAmount,
+		message_length: usize,
+	) -> GasAmount {
+		use crate::eth::fees::*;
+		// Adding one extra gas unit per message's length (byte) for the extra gas overhead of
+		// passing the message through the Vault. The extra l2 gas per message's calldata byte
+		// should be included in the user's gas budget.
+		(gas_budget
+			.saturating_add(if is_native_asset {
+				CCM_VAULT_NATIVE_GAS_OVERHEAD
+			} else {
+				CCM_VAULT_TOKEN_GAS_OVERHEAD
+			})
+			.saturating_add(message_length as u128))
```

Comment:
Here we are only calculating the gas limit, which is the user's specified gas budget + gas overhead. In Arbitrum the calculation is way more complex because the gas limit depends on a bunch of factors like the `L1_GAS_PER_BYTES`.
The Ethereum gas per bytes is part of the user's specified gas budget. It's a bit cumbersome.